staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/structural.go

```diff
@@ -92,6 +92,29 @@ type Extensions struct {
 	//      - type: string
 	//    - ... zero or more
 	XIntOrString bool
+
+	// x-kubernetes-list-map-keys annotates map lists by specifying the keys used
+	// as the index of the map.
+	//
+	// This tag MUST only be used on lists that have the "x-kubernetes-list-type"
```

Comment:
what does the "MUST only" mean?

Reply:
If there are constraints around these fields, they must be validated in the CRD or structural validation. And it needs test in validation_test.go.